src/Shared/FileMatcher.cs

```diff
@@ -2499,13 +2499,13 @@ private static string[] CreateArrayWithSingleItemIfNotExcluded(string filespecUn
             catch (AggregateException ex)
             {
                 // Flatten to get exceptions than are thrown inside a nested Parallel.ForEach
-                if (ex.Flatten().InnerExceptions.All(ExceptionHandling.IsIoRelatedException))
+                if (ex.Flatten().InnerExceptions.All(ex => ExceptionHandling.IsIoRelatedException(ex) && !(ex is FileLoadException)))
                 {
                     return CreateArrayWithSingleItemIfNotExcluded(filespecUnescaped, excludeSpecsUnescaped);
                 }
                 throw;
             }
-            catch (Exception ex) when (ExceptionHandling.IsIoRelatedException(ex))
+            catch (Exception ex) when (ExceptionHandling.IsIoRelatedException(ex) && !(ex is FileLoadException))
```

Comment:
I am not quite sure here should I or should I not move this change in `IsIoRelatedException` function. I feel should be possible, but I would prefer to keep it here just in case.

Reply:
Decided to move that to ManagedFileSystem class.